test/agentchat/contrib/test_pgvector_retrievechat.py

```diff
@@ -23,18 +24,21 @@
     from autogen.agentchat.contrib.retrieve_user_proxy_agent import (
         RetrieveUserProxyAgent,
     )
+
+    PGVECTOR_INSTALLED = True
 except ImportError:
     skip = True
+    PGVECTOR_INSTALLED = False
 else:
-    skip = False
+    skip = False or skip_openai
 
 
 test_dir = os.path.join(os.path.dirname(__file__), "../..", "test_files")
 
 
 @pytest.mark.skipif(
-    sys.platform in ["darwin", "win32"] or skip,
-    reason="do not run on MacOS or windows OR dependency is not installed OR requested to skip",
+    not PGVECTOR_INSTALLED or skip,
```

Comment:
It seems that we only need one of PGVECTOR_INSTLLED or skip, not both.

Reply:
I removed the duplicate condition.